packages/serverless-contracts/src/contracts/apiGateway/features/lambdaTrigger.ts

```diff
@@ -0,0 +1,49 @@
+import { ApiGatewayContract } from '../apiGatewayContract';
+import {
+  ApiGatewayLambdaCompleteTriggerType,
+  ApiGatewayLambdaConfigType,
+  ApiGatewayLambdaSimpleTriggerType,
+  ApiGatewayTriggerKey,
+} from '../types';
+
+/**
+ * Returns a basic serverless function trigger associated to an ApiGatewayContract
+ *
+ * @param contract your ApiGatewayContract
+ */
+export const getTrigger = <Contract extends ApiGatewayContract>(
+  contract: Contract,
+): ApiGatewayLambdaSimpleTriggerType<
+  ApiGatewayTriggerKey<Contract['integrationType']>
+> => {
+  const key = contract.integrationType === 'httpApi' ? 'httpApi' : 'http';
+
+  // @ts-ignore somehow the type inference does not work here
+  return { [key]: { path: contract.path, method: contract.method } };
+};
+
+/**
+ * Returns an enhanced serverless function trigger associated to an ApiGatewayContract
+ *
+ * @param contract your ApiGatewayContract
+ * @param additionalConfig for example an authorizer reference, ...
+ */
+export const getCompleteTrigger = <Contract extends ApiGatewayContract>(
+  contract: Contract,
+  additionalConfig: ApiGatewayLambdaConfigType<
```

Comment:
These two functions now seem very similar and could maybe be merged in a single `getTrigger` function with an optional `additionalConfig` argument